framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -942,10 +942,18 @@ public AnnotatedTypeMirror getAnnotatedType(Tree tree) {
             return treeCache.get(tree).deepCopy();
         }
 
-        AnnotatedTypeMirror type;
         if (TreeUtils.isClassTree(tree)) {
-            type = fromClass((ClassTree)tree);
-        } else if (tree.getKind() == Tree.Kind.METHOD ||
+            ClassTree classTree = (ClassTree) tree;
+            AnnotatedTypeMirror type = getAnnotatedTypeOfClassTree(classTree);
```

Comment:
This method also checks whether the tree is in the cache, which already happened on line 941.